src/providers/FusionAuthProvider.tsx

```diff
@@ -1,95 +1,133 @@
-import React, { useCallback, useContext, useMemo } from 'react';
+import React, {
+    useCallback,
+    useContext,
+    useEffect,
+    useMemo,
+    useState,
+} from 'react';
 import { TextEncoder } from 'util';
+import axios from 'axios';
+import Cookies from 'js-cookie';
 
 export interface IFusionAuthContext {
     login: (state: string) => Promise<void>;
     logout: () => Promise<void>;
     register: (state: string) => Promise<void>;
+    user: Record<string, any>;
 }
 
 export const FusionAuthContext = React.createContext<IFusionAuthContext>({
     login: () => Promise.resolve(),
     logout: () => Promise.resolve(),
     register: () => Promise.resolve(),
+    user: {},
 });
 
-interface Props {
+export interface FusionAuthConfig {
     baseUrl: string;
     clientID: string;
+    serverUrl: string;
     scope: string;
     redirectUri: string;
     idTokenHint?: string;
+}
+
+interface Props {
+    config: FusionAuthConfig;
     children?: React.ReactNode;
 }
 
-export const FusionAuthProvider: React.FC<Props> = ({
-    baseUrl,
-    clientID,
-    scope,
-    redirectUri,
-    idTokenHint,
-    children,
-}) => {
+export const FusionAuthProvider: React.FC<Props> = ({ config, children }) => {
+    const [user, setUser] = useState<Record<string, any>>({});
+
     const generateUrl = useCallback(
         (functionType: FunctionType, queryParams: Record<string, string>) => {
             const query = new URLSearchParams(queryParams);
 
-            return `${baseUrl}/oauth2/${functionType}?${query}`;
+            return `${config.baseUrl}/oauth2/${functionType}?${query}`;
         },
-        [baseUrl],
+        [config],
     );
 
     const login = useCallback(
         async (state = '') => {
+            const rand = generateRandomString();
+            Cookies.set('lastState', rand);
```

Comment:
We should be storing the entire state string `rand:state` not just the random portion. Otherwise the comparison when we check the state value will always fail